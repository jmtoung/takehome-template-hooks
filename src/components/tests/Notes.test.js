// import React from 'react';
// import Notes from '../Notes';
// import { shallow } from 'enzyme';
// import toJSON from 'enzyme-to-json';

// const data = {
//   total: 3,
//   notes: [
//     {id: 1, title: 'title1', body: 'body1'},
//     {id: 2, title: 'title2', body: 'body2'},
//     {id: 3, title: 'title3', body: 'body3'}
//   ]
// };

// describe('Notes', () => {
//   it('displays notes from state', () => {
//     const wrapper = shallow(<Notes page='1' />);
//     wrapper.setState(data);
//     expect(toJSON(wrapper)).toMatchSnapshot();
//   });

//   it('gets notes from api.', async (done) => {
//     const mockSuccessResponse = data;
//     const mockJsonPromise = Promise.resolve(mockSuccessResponse);
//     const mockFetchPromise = Promise.resolve({
//       json: () => mockJsonPromise,
//     });
//     jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise);

//     const wrapper = shallow(<Notes page='1' />);
//     expect(wrapper.state()).toEqual({
//       total: 0,
//       page: "1",
//       notes: []
//     });

//     expect(global.fetch).toHaveBeenCalledTimes(1);
//     expect(global.fetch).toHaveBeenCalledWith('https://5ec581652a4ba000163d2084.mockapi.io/api/v1/notes?page=1&limit=10');

//     process.nextTick(() => {
//       expect(wrapper.state()).toEqual({
//         total: 3,
//         page: "1",
//         notes: [
//           {id: 1, title: 'title1', body: 'body1'},
//           {id: 2, title: 'title2', body: 'body2'},
//           {id: 3, title: 'title3', body: 'body3'}
//         ]
//       });
//     });

//     global.fetch.mockClear();
//     done();
//   });
// });