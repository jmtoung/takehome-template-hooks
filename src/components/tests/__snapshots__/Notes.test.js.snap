// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notes displays notes from state 1`] = `
<div>
  <div>
    <ul
      className="pagination"
    >
      <li
        className="page-item"
      >
        <Link
          className="page-link"
          to="/"
        >
          First
        </Link>
      </li>
      <li
        className="page-item"
        key="1"
      >
        <Link
          className="page-link"
          to="/1"
        >
          1
        </Link>
      </li>
      <li
        className="page-item"
      >
        <Link
          className="page-link"
          to="/1"
        >
          Last
        </Link>
      </li>
    </ul>
  </div>
  <div>
    <ul>
      <Note
        body="body1"
        deleteNote={[Function]}
        id={1}
        key="1"
        title="title1"
      />
      <Note
        body="body2"
        deleteNote={[Function]}
        id={2}
        key="2"
        title="title2"
      />
      <Note
        body="body3"
        deleteNote={[Function]}
        id={3}
        key="3"
        title="title3"
      />
    </ul>
  </div>
</div>
`;
